### DISCUSSION TITLE
Changing rates... no need to assume CAKE qdisc

### DISCUSSION THREAD

**Discussion by dlakelan** [sqm-autorate/sqm-autorate#116] [Algorithm Design]
I kinda think we should outsource the changing rates to a small shell script... something like:

```
change_rate eth0 12000
```
which would internally decide which tc commands were needed, so we could handle qdisc setups using TBF or HTB or HFSC or whatever

**Comment by chas-iot** [comment]
great idea so that we can live up to our project title of _sqm-autorate_ rather than _cake-autorate_

I suggest that we do it in little pluggable lua modules rather than shell to reduce the cognitive load - but I'm not tied to that idea

**Reply by dlakelan** [reply]
> pluggable lua modules rather than shell to reduce the cognitive load

I'm not opposed to this, but I do think the lua modules will probably read something like:

```
local ex = os.execute
ex("...")
ex("...")
ex("...")
```
which is to say... they'll effectively be shell scripts.
